optimizely/optimizely_factory.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2021, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from . import logger as optimizely_logger
+from .config_manager import PollingConfigManager
+from .error_handler import NoOpErrorHandler
+from .event.event_processor import BatchEventProcessor
+from .event_dispatcher import EventDispatcher
+from .notification_center import NotificationCenter
+from .optimizely import Optimizely
+
+
+class OptimizelyFactory(object):
+    """ Optimizely factory to provides basic utility to instantiate the Optimizely
+        SDK with a minimal number of configuration options."""
+
+    max_event_batch_size = None
+    max_event_flush_interval = None
+    polling_interval = None
+    blocking_timeout = None
+
+    @staticmethod
+    def set_batch_size(batch_size, logger=optimizely_logger.NoOpLogger().logger):
```

Comment:
passing a logger to set a parameter sounds weird. Can't remove all these validation with logging here since those parameters are validated again in event_processor.